packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1109,11 +1109,19 @@ class CrudStoreImpl
   }
 
   async openBulkUpdateModal() {
+    // This will be set if the user clicked a bulk update query on the My Queries page
+    const initialUpdate: BSONObject | undefined = (this.options.query as any)
+      ?.update;
+    const updateText = initialUpdate
+      ? toJSString(initialUpdate) || INITIAL_BULK_UPDATE_TEXT
+      : INITIAL_BULK_UPDATE_TEXT;
```

Comment:
Super nit, but maybe would be easier to split this logic in two parts. The function here can take an optional argument:

```suggestion
  async openBulkUpdateModal(updateText = INITIAL_BULK_UPDATE_TEXT) {
```

And the place where it's called below can stringify the update from options? I feel like that way you don't need to explain here why options will have the `query` property at all.

No strong preference though!

Reply:
Aah that's much better and I don't know why I didn't think about it. Will move it.